maven-java/kalix-maven-plugin/src/main/java/kalix/DeployMojo.java

```diff
@@ -36,32 +35,51 @@ public class DeployMojo extends AbstractMojo {
     @Parameter(property = "kalixContext")
     private String kalixContext;
 
+    @Parameter(property = "kalixProject")
+    private String kalixProject;
+
     @Parameter(defaultValue = "30000", property = "cliTimeoutMs", required = true)
     private Long cliTimeoutMs;
 
     private final Log log = getLog();
 
     /**
-     * We deploy by invoking the services deploy command
+     * We deploy by invoking the services deploy command to 'kalixProject' if set.
+     * If 'kalixProject' is not set then deploying to the currently selected project.
      */
     public void execute() throws MojoExecutionException {
-        log.info("Deploying project to Kalix");
+        final List<String> commandLine;
+        final int deploymentResult = 0;
+        if (kalixProject.isEmpty()) {
+            log.info("`kalixProject` hasn't been set. Therefore, deploying to the currently selected project configured via Kalix CLI");
+            return;
+        }
+        if (kalixContext != null) {
+            commandLine = Arrays.asList(kalixPath, "--context", kalixContext, "--project", kalixProject, "service", "deploy", service, dockerImage);
+            deploy(commandLine);
+        } else {
+            commandLine = Arrays.asList(kalixPath, "--project", kalixProject, "service", "deploy", service, dockerImage);
+            deploy(commandLine);
+        }
+    }
+
+    private void deploy(List<String> commandLine) throws MojoExecutionException {
         try {
-            final List<String> commandLine;
-            if (kalixContext != null) {
-                commandLine = Arrays.asList(kalixPath, "--context", kalixContext, "service", "deploy", service, dockerImage);
-            } else {
-                commandLine = Arrays.asList(kalixPath, "service", "deploy", service, dockerImage);
-            }
+            log.info("Deploying project to Kalix");
             log.info("Executing `" + String.join(" ", commandLine) + "`");
             Process process = new ProcessBuilder().directory(baseDir).command(commandLine).start();
             synchronized (process) {
                 process.wait(cliTimeoutMs);
             }
-            int status = process.exitValue();
-            if (status == 0) {
+            final int deploymentResult = process.exitValue();
+            if (deploymentResult == 0) {
                 log.info("Done.");
```

Comment:
Would be nice if this was more informative, something like "Successfully deployed Kalix service [service] in [project] (context: [context])"

Could be made easier if the method was `deploy(String service, Optional<String> project, Optional<String> context, String dockerImage)` and turning that to array was done in here

Reply:
added